server-adapters/resteasy-undertow-spring/src/test/java/org/jboss/resteasy/springmvc/test/client/BasicSpringTest.java

```diff
@@ -0,0 +1,67 @@
+package org.jboss.resteasy.springmvc.test.client;
+
+import io.undertow.servlet.api.DeploymentInfo;
+import org.apache.commons.httpclient.HttpException;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;
+import org.jboss.resteasy.client.jaxrs.internal.ResteasyClientBuilderImpl;
+import org.jboss.resteasy.plugins.server.undertow.spring.UndertowJaxrsSpringServer;
+import org.jboss.resteasy.test.TestPortProvider;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class BasicSpringTest {
+
+   UndertowJaxrsSpringServer server;
+
+   @Before
+   public void before() throws Exception {
+      server = new UndertowJaxrsSpringServer();
+
+      server.start();
+
+      DeploymentInfo deployment = server.undertowDeployment("classpath:spring-servlet.xml", null);
+      deployment.setDeploymentName(BasicSpringTest.class.getName());
+      deployment.setContextPath("/");
+      deployment.setClassLoader(BasicSpringTest.class.getClassLoader());
+
+      server.deploy(deployment);
+   }
+
+   @After
+   public void after() {
+      server.stop();
+   }
+
+   @Test
+   public void testBasic() throws HttpException, IOException {
+      ResteasyClient client = new ResteasyClientBuilderImpl().build();
```

Comment:
While you're here, consider changing this line and the following one to use the JAX-RS api instead of the RESTEasy one (so ClientBuilder instead of ResteasyClientBuilder, WebTarget instead of ResteasyWebTarget, as no RESTEasy specific method seems to be needed.

Reply:
@asoldano this test is using the resteasy client proxy framework:

```java
BasicResource br = target.proxy(BasicResource.class);
```

So IMHO we can keep it. For the ones used in `AsynchSpringTest`, I have switched to jaxrs standard classes. wdyt?

Reply:
Ok, fine, actually, the remaining improvement here could have been to create the builder using standard api and then cast it to ResteasyClientBuilder to avoid direct reference to an internal/implementation detail class (ResteasyClientBuilderImpl), but it's not something worth another change here.

Reply:
But still the client should be closed ;-)

Reply:
yes sir!